File: 2020/Day 02/1.py
Array = []
for Ln in open("input.txt"):
    N, C, P = Ln.split(); C = C[0]
    N = (*map(int, N.split('-')),)
    Array.append((N, C, P))

print("Silver:", sum([1 for (Min, Max), Ch, Pwd
    in Array if Min <= Pwd.count(Ch) <= Max]))

print("Gold:", sum([1 for Pos, Ch, Pwd in Array if
    sum([1 for I in Pos if Pwd[I-1] == Ch]) == 1]))

File: 2020/Day 03/1.rb
I, V = File.readlines('input.txt'),
  [[1, 1], [3, 1], [5, 1], [7, 1], [1, 2]]

def j(a, b)
  (0..I.count-1).step(b).map{|y|
    I[y][a * y/b % 31]}.count(?#)
end

puts "Silver: #{j(*V[1])}",
  "Gold: #{V.inject(1){j(*_2)*_1}}"
